p2p/host/basic/basic_host.go

```diff
@@ -469,26 +469,41 @@ func (h *BasicHost) Addrs() []ma.Multiaddr {
 	return h.addrs(h.AllAddrs())
 }
 
+// mergeAddrs merges input address lists, leave only unique addresses
+func mergeAddrs(addrLists ...[]ma.Multiaddr) (uniqueAddrs []ma.Multiaddr) {
+	exists := make(map[string]bool)
+	for _, addrList := range addrLists {
+		for _, addr := range addrList {
+			if exists[addr.String()] {
```

Comment:
We can actually just call `string(addr)`. calling `addr.String()` will format the address as a string (e.g., `/ip4/.../tcp/.../ipfs/...`) which will be significantly more expensive than just converting the bytes representation to a string.

Reply:
Basically, `addr.String()` requires a bunch of allocations and complicated logic while `string(thing)` is almost free (at most one allocation).